### PR TITLE
Add Joining Tables, Update Data and Query Linked Tables

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
- In this PR I added joining tables, inserted more data into tables and queried multiple tables.

**Highlights**

- Practice many-to-many relational database modelling.
- Use primary keys and foreign keys for joining tables.
- Insert data into vets, species and visits tables;
- Query multiple tables  to answer analytical questions

Below are the screenshots of the results of my queries.

**Cheers and Happy Reviewing!**
Capture21
Capture22
Capture23
Capture24
Capture25
Capture26
Capture27
Capture28

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -36,3 +36,24 @@ ON DELETE CASCADE;
...
  ADD COLUMN owners_id INT REFERENCES owners (id) 
  ON DELETE CASCADE;
+
+CREATE TABLE vets (
+    id INT NOT NULL GENERATED ALWAYS AS IDENTITY, 
+    name text NOT NULL, 
+    age INT NOT NULL, 
+    date_of_graduation date,
+    PRIMARY KEY (id)
+    );
```
Comment (mateo951) [inline schema.sql:46]: Add a new table with the name vets with the propper columns ✔

File: schema.sql [hunk 2]
```diff
@@ -36,3 +36,24 @@ ON DELETE CASCADE;
...
+
+CREATE TABLE specializations (
+    id INT NOT NULL GENERATED ALWAYS AS IDENTITY,
+    species_id INTEGER REFERENCES species (id), 
+    vets_id INTEGER REFERENCES vets (id)
+    );
```
Comment (mateo951) [inline schema.sql:52]: Add a new join table with the name `specializations` 📝

File: data.sql [hunk 3]
```diff
@@ -50,3 +50,34 @@ UPDATE animals SET owners_id = (SELECT id FROM owners WHERE full_name = 'Melody
...
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(1, 2, '07/04/2021');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(4, 3, '29/09/2019');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(2, 4, '03/10/2020');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(2, 4, '04/11/2020');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(8, 2, '24/01/2019');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(8, 2, '15/01/2019');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(8, 2, '27/02/2020');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(8, 2, '03/08/2020');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(10, 1, '24/05/2020');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES(10, 1, '11/05/2021');
```
Comment (mateo951) [inline data.sql:83]: Well done 🙌 Expected data was introduced 😎

File: queries.sql [hunk 4]
```diff
@@ -43,3 +43,15 @@ SELECT full_name, STRING_AGG (animals.name, ', ') AS animals FROM owners LEFT JO
...
+SELECT animals.name,COUNT(animals.name) FROM animals INNER JOIN visits ON animals.id = visits.animals_id GROUP BY animals.name ORDER BY COUNT(animals.name) DESC LIMIT 1;
+SELECT animals.name FROM animals INNER JOIN visits ON visits.animals_id = animals. id INNER JOIN vets ON visits.vets_id = vets.id WHERE vets.id = 2 AND date_of_visit = (SELECT MIN (date_of_visit) FROM visits);
+SELECT * FROM animals INNER JOIN visits ON visits.animals_id = animals. id INNER JOIN vets ON visits.vets_id = vets.id WHERE date_of_visit = (SELECT MAX (date_of_visit) FROM visits);
+SELECT COUNT(visits.animals_id) FROM visits INNER JOIN vets ON visits.vets_id = vets.id INNER JOIN animals ON animals.id = visits.animals_id INNER JOIN specializations ON specializations.species_id = vets.id WHERE specializations.species_id != animals.species_id;
+SELECT species.name, COUNT(*) FROM animals INNER JOIN visits ON animals.id = visits.animals_id INNER JOIN species ON animals.species_id = species.id WHERE visits.vets_id = 2 GROUP BY species.name ORDER BY COUNT(*) DESC LIMIT 1;
```
Comment (mateo951) [inline queries.sql:55]: All questions were answered 🤓

**Review by mateo951 — APPROVED** [review]
### Hi @sharon-odhiambo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sharon-odhiambo** [comment]
Hello @mateo951 , thank you for the review plus the inline comments!

**Pull request merged**